google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -965,13 +975,7 @@ def listen &callback
         ##
         # @private Start a new Query.
         def self.start query, parent_path, client, limit_type: nil
-          query ||= StructuredQuery.new
-          Query.new.tap do |q|
-            q.instance_variable_set :@query, query
-            q.instance_variable_set :@parent_path, parent_path
-            q.instance_variable_set :@limit_type, limit_type
-            q.instance_variable_set :@client, client
-          end
+          new query, parent_path, client, limit_type: limit_type
```

Comment:
Thanks for getting rid of all the `instance_variable_set`s. Those felt so dirty...